src/Storages/MergeTree/IExecutableTask.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <memory>
+#include <functional>
+
+#include <common/shared_ptr_helper.h>
+#include <Interpreters/StorageID.h>
+
+namespace DB
+{
+
+class IExecutableTask
+{
+public:
+    virtual bool execute() = 0;
```

Comment:
From reader's point of view `bool execute()` means: `return true if success and false if not successful`. But here it means `step finished, need execute one more time`. Add a comment. Maybe `executeStep`?

Reply:
Renamed.